Convert docs source to markdown

mosabua opened the pull request:
## Description

Continued conversion of docs source to markdown/myst

- All files in language folder
- All files in sql folder
- Adjust test for reserved keyword list

Verified that the output HTML is identical or only differs in whitespace, wrapping and other aspects that do not impact the rendered, visible output.

## Additional context and related issues

#17732 
#17742 
#17759

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

ebyhr reviewed (changes requested):
Some pages are broken. e.g. 
<img width="835" alt="Screenshot 2023-07-25 at 12 16 45" src="https://github.com/trinodb/trino/assets/6237050/865f5a53-5472-40db-939a-4eefd7405d4d">

mosabua commented:
> Some pages are broken. e.g. <img alt="Screenshot 2023-07-25 at 12 16 45" width="835" src="https://user-images.githubusercontent.com/6237050/255778105-865f5a53-5472-40db-939a-4eefd7405d4d.png">

Whoa .. thats weird.. worked in my testing. Let me check some more.

mosabua commented:
Fixed the issues .. the comment for START inside the table for some reason did not work in new testing .. the sphinx iterative builds are a pain sometimes.. I also adjusted the test .. all should work now

mosabua commented:
Build failure is an unrelated Iceberg test.

ebyhr commented on core/trino-parser/src/main/java/io/trino/sql/ReservedIdentifiers.java:
```diff
@@ -71,28 +72,22 @@ private static void validateDocs(Path path)
         while (true) {
             String line = iterator.next();
-            if (line.startsWith(TABLE_PREFIX)) {
+            if (line.startsWith(TABLE_END)) {
```
There's no need to put dummy line `% END`. We can break the loop when the line doesn't start with `|`.

mosabua replied:
Well .. I thought its better to be explicit rather than implicit .. but fine .. I will change.

ebyhr commented on core/trino-parser/src/main/java/io/trino/sql/ReservedIdentifiers.java:
```diff
@@ -39,8 +39,9 @@
-    private static final String TABLE_PREFIX = "============================== ";
+    private static final Pattern TABLE_ROW = Pattern.compile("\\| `([A-Z_]+).*");
+    private static final String TABLE_START = "| ------------------- |";
+    private static final String TABLE_END = "% END";
```
This constant can be removed. Please see another comment.

mosabua commented:
Also adjusted for merge of https://github.com/trinodb/trino/pull/18431

electrum commented on core/trino-parser/src/main/java/io/trino/sql/ReservedIdentifiers.java:
```diff
@@ -39,8 +39,9 @@
     private static final Pattern IDENTIFIER = Pattern.compile("'([A-Z_]+)'");
-    private static final Pattern TABLE_ROW = Pattern.compile("``([A-Z_]+)``.*");
-    private static final String TABLE_PREFIX = "============================== ";
+    private static final Pattern TABLE_ROW = Pattern.compile("\\| `([A-Z_]+).*");
```
Match the closing quote of the keyword, just to be safe and consistent
```suggestion
    private static final Pattern TABLE_ROW = Pattern.compile("\\| `([A-Z_]+)` .*");
```

mosabua replied:
Updated
